syntax = "proto3";

option java_multiple_files = true;
option java_package = "ua.edu.ucu";
option java_outer_classname = "LoggerService";

service Logger {
  rpc AppendMessage (AppendMessageRequest) returns (AppendMessageResponse) {}
  rpc ListMessages (ListMessagesRequest) returns (ListMessagesResponse) {}
}

message LogMessage {
  string log = 1;
}

message AppendMessageRequest {
  LogMessage log = 1;
}

message AppendMessageResponse {
  AppendResponseCode responseCode = 1;
}

enum AppendResponseCode {
  OK = 0;
  SOME_ERROR = 1;
}

message ListMessagesRequest {
}

message ListMessagesResponse {
  repeated LogMessage logs = 1;
}